lib/MultiTopicsConsumerImpl.cc

```diff
@@ -886,3 +906,8 @@ void MultiTopicsConsumerImpl::messageProcessed(Message& msg) {
 std::shared_ptr<MultiTopicsConsumerImpl> MultiTopicsConsumerImpl::get_shared_this_ptr() {
     return std::dynamic_pointer_cast<MultiTopicsConsumerImpl>(shared_from_this());
 }
+
+void MultiTopicsConsumerImpl::beforeConnectionChange(ClientConnection& cnx) {
+    consumers_.forEachValue(
+        [&cnx](const ConsumerImplPtr& consumer) { consumer->beforeConnectionChange(cnx); });
```

Comment:
For `MultiTopicConsumerImpl`, It shouldn't be connected, right? 

Should it be more reasonable to change this implementation to throw an exception or add an empty implementation to `ConsumerImplBase`?

Reply:
Good suggestion, adopted.